lib/sign.js

```diff
@@ -6,13 +6,31 @@ const chalk = require('chalk')
 
 const utils = require('./utils')
 
-function makeCert (publisherName, certFilePath, program) {
-  const cer = path.join(certFilePath, `${publisherName}.cer`)
-  const pvk = path.join(certFilePath, `${publisherName}.pvk`)
-  const pfx = path.join(certFilePath, `${publisherName}.pfx`)
+// MakeCert supports publisher strings matching the (X.500 distinguished name) regex below
+// See https://msdn.microsoft.com/en-us/library/windows/apps/br211441.aspx
+const validPublisherRegex = /(CN|L|O|OU|E|C|S|STREET|T|G|I|SN|DC|SERIALNUMBER|(OID\.(0|[1-9][0-9]*)(\.(0|[1-9][0-9]*))+))=(([^,+="<>#;])+|".*")(, ((CN|L|O|OU|E|C|S|STREET|T|G|I|SN|DC|SERIALNUMBER|(OID\.(0|[1-9][0-9]*)(\.(0|[1-9][0-9]*))+))=(([^,+="<>#;])+|".*")))*/
 
-  const makecert = path.join(program.windowsKit, 'makecert.exe')
-  const makecertArgs = ['-r', '-h', '0', '-n', `CN=${publisherName}`, '-eku', '1.3.6.1.5.5.7.3.3', '-pe', '-sv', pvk, cer]
+function isValidPublisher (publisherName) {
```

Comment:
Actually, I think `isValidPublisherName` is more appropriate here, as long as the verbosity isn't too annoying to people.